src/query/interpreter.cpp

```diff
@@ -327,20 +336,26 @@ class ReplQueryHandler final : public query::ReplicationQueryHandler {
     }
   }
 
-  /// @throw QueryRuntimeException if an error ocurred.
-  void DropReplica(const std::string &replica_name) override {
-    if (db_->GetReplicationRole() == storage::replication::ReplicationRole::REPLICA) {
-      // replica can't unregister a replica
-      throw QueryRuntimeException("Replica can't unregister a replica!");
-    }
-    if (!db_->UnregisterReplica(replica_name)) {
-      throw QueryRuntimeException(fmt::format("Couldn't unregister the replica '{}'", replica_name));
+  /// @throw QueryRuntimeException if an error occurred.
+  void DropReplica(std::string_view replica_name) override {
+    auto const result = handler_.UnregisterReplica(replica_name);
+    switch (result) {
+      using enum memgraph::storage::UnregisterReplicaResult;
+      case NOT_MAIN:
+        throw QueryRuntimeException("Replica can't unregister a replica!");
+      case COULD_NOT_BE_PERSISTED:
+        [[fallthrough]];
```

Comment:
Why not throw here separate QueryRuntimeException?

Reply:
Refactor for now, maybe in future we will add another QueryRuntimeException.